crates/circuit/src/operations.rs

```diff
@@ -2087,6 +2088,27 @@ pub struct PyInstruction {
     pub instruction: PyObject,
 }
 
+impl PyInstruction {
+    pub fn blocks(&self) -> Option<impl Iterator<Item = CircuitData>> {
+        Python::with_gil(|py| -> Option<vec::IntoIter<CircuitData>> {
+            let raw_blocks = self.instruction.getattr(py, "blocks").ok()?;
```

Comment:
At the point of creating a `PyInstruction`, we type-check and cache whether it's a proper `ControlFlowOp` (stored in `PyInstruction::control_flow`).  We ought to check that field before doing this, because there's nothing stopping a user-defined `Instruction` from having a `blocks` attribute that's a `tuple`, but not what we expect.

Reply:
Done in 591b0cba56737425a2df94cae65c593d89fde389.